Examples/Framework/include/ActsExamples/Framework/WhiteBoard.hpp

```diff
@@ -99,17 +84,16 @@ inline const T& ActsExamples::WhiteBoard::get(const std::string& name) const {
   if (it == m_store.end()) {
     throw std::out_of_range("Object '" + name + "' does not exists");
   }
-  const IHolder* holder = it->second.get();
 
-  const auto* castedHolder = dynamic_cast<const HolderT<T>*>(holder);
-  if (castedHolder == nullptr) {
-    throw std::out_of_range("Type mismatch for object '" + name + "'");
+  try {
+    const T& ret = *std::any_cast<T>(&it->second);
```

Comment:
I think the pointer form of `std::any_cast<T>(const std::any*)` returns `nullptr` if the types don't match (case 4, [here](https://en.cppreference.com/w/cpp/utility/any/any_cast)). It doesn't throw `std::bad_any_cast` as the non-pointer form does.
The behaviour is like what `dynamic_cast` does, so this is closer to the original code.

Reply:
Ah you're right!